drafts/AENS.md

```diff
@@ -0,0 +1,414 @@
+
+Giving users plenty time to prepare for a launch of the naming
+system is of utmost importance given the allocation rules, i.e.
+first come first serve, of this first draft.
+This also includes giving users the proper tools, i.e. a GUI,
+in order to level the playing field as to who is able to actually 
+register names.
+
+Since we don't have an auction protocol just yet, we could hold
+auctions on Ethereum to establish an initial allocation of names, 
+just like our ERC-20 token.
+
+
+## (possible) Future extensions
+
+### Registrars
+
+A later version of the name service should allow every user
+to become a registrar and encode their registration logic
+in a smart contract. A simplification would be that registrars
+have to submit the claim transaction themselves and just
+set the owner to whoever they want to give the name to.
+
+
+### Auctions
+
+Using (Vickrey) auctions instead of a simple, first come
+first serve approach for new claims, might allow a better
+mechanism for price discovery without having to go through
+a secondary market.
+
+
+### URI Schema
+
+We are currently using the same schema as DNS, which might
+be problematic since this comes with a big slew of expectations
+as to what can and can't be done.
+
+See [this post about problems of urls](https://noncombatant.org/2017/11/07/problems-of-urls/)
+or [this IPFS spec issue](https://github.com/ipfs/specs/pull/152) for
+some discussions.
+
+### Decentralized name exchange
+
+Having the exchange for names on chain would enable us to
+transfer names in a trustless manner. Plus having the bids
+recorded on chain could also facilitate better price discovery.
+
+
+### Governance
+
+Allowing users to register their own namespaces and become
+registrars themselves, i.e. if I own `mywallet.aet`, I can
+then allow others to register `name.mywallet.aet`.
+
+Consider a voting mechanism to introduce new TLDs.
+
+
+### Interoperability
+
+- investigate possibility to integrate with ENS
+- add interoperability with https://ipld.io
+- add support for https://multiformats.io/multihash/
+
+
+### Light clients
+
+A light client might only be interested in the history or
+integrity particular name and thus require some lightweight 
+mechanism for this, which is currently not possible.
+
+
+### Fee lottery
+
+It was suggested that the fees for the `.aet` namespace could
+be distributed to random accounts via a lottery.
+
+## Open Questions:
+
+(***TODO***: Pedersen commitments, i.e. perfectly hiding, or hash commitment? 
+This also has implications for the allocation mechanism. With a simple hash
+commitment you could enable auctions on hashes, e.g. what ENS does, where people
+who are interested in a name can hold an auction without revealing the actual
+name. This would not be possible with Pedersen commitmens. On the other
+hand in the first come first serve case with hash commitments, miners could 
+keep a dictionary of names they want to snatch up whenever somebody tries to 
+register them)
```

Comment:
Would it make sense to 'salt' the namehash in the pre-commit message? Then the miners can't keep a table.

Reply:
That certainly would be possible but then we'd lose the ability to hold auctions on the hashes. Though, with auctions, the miners wouldn't be able to front run anymore anyway.

Reply:
Maybe the salt could be public (part of the pre-commit message) we are just trying to avoid pre-computed tables in miners...